l10n_ro_account_spv/models/message_spv.py

```diff
@@ -0,0 +1,357 @@
+# Copyright (C) 2024 Dorin Hongu <dhongu(@)gmail(.)com
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import io
+import logging
+import zipfile
+from base64 import b64encode
+
+import requests
+from lxml import etree
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class MessageSPV(models.Model):
+    _name = "l10n.ro.message.spv"
+    _description = "Message SPV"
+    _order = "date desc"
+
+    name = fields.Char(string="Message ID")  # id
+    cif = fields.Char(string="CIF")  # cif
+    message_type = fields.Selection(
+        [
+            ("in_invoice", "In Invoice"),
+            ("out_invoice", "Out Invoice"),
+            ("error", "Error"),
+        ],
+        string="Type",
+    )  # tip
+    date = fields.Datetime(string="Date")  # data_creare
+    details = fields.Char(string="Details")  # detalii
+    error = fields.Text(string="Error")  # eroare
+    request_id = fields.Char(string="Request ID")  # id_solicitare
+    ref = fields.Char(string="Reference")  # referinta
+
+    # campuri suplimentare
+
+    invoice_id = fields.Many2one("account.move", string="Invoice")
+    partner_id = fields.Many2one("res.partner", string="Partner")
+
+    # draft - starea initiala a mesajului descarcat din SPV
+    # downloaded - fisierul a fost descarcat cu succes
+    # invoice - factura a fost creata cu succes
+    # done - factura a fost creata si validata cu succes
+    state = fields.Selection(
+        [
```

Comment:
Poate sa fie mesaj in draft, odata ce e descarcat, invoice nu cred ca are rost atata timp catvai invoice_id, cred ca e destul downloaded,  attached si mapped

Reply:
attached si mapped cand sa fie setate ?

Reply:
Attached cand e descarcat atasamentul, mapped cand are factura linkuita